core/src/main/java/io/questdb/griffin/HttpSqlExecutionInterruptor.java

```diff
@@ -0,0 +1,69 @@
+package io.questdb.griffin;
+
+import java.io.Closeable;
+
+import io.questdb.cairo.CairoException;
+import io.questdb.network.NetworkFacade;
+import io.questdb.std.Unsafe;
+
+public class HttpSqlExecutionInterruptor implements SqlExecutionInterruptor, Closeable {
+    private final NetworkFacade nf;
+    private final int nIterationsPerCheck;
+    private final int bufferSize;
+    private long buffer;
+    private int nIterationsSinceCheck;
+    private long fd = -1;
+
+    public HttpSqlExecutionInterruptor(NetworkFacade nf, int nIterationsPerCheck, int bufferSize) {
+        super();
+        this.nf = nf;
+        this.nIterationsPerCheck = nIterationsPerCheck;
+        this.bufferSize = bufferSize;
+        buffer = Unsafe.malloc(bufferSize);
+    }
+
+    @Override
+    public void checkInterrupted() {
+        assert fd != -1;
+        if (nIterationsSinceCheck == nIterationsPerCheck) {
```

Comment:
I can change this, but I dont agree with it. I doubt it will have any significant impact and may actually be slower on more modern processors which are biased towards the if statement being true regardless of the compiler hints